js/scripts.js

```diff
@@ -0,0 +1,8 @@
+var list = document.getElementById('list');
+var addElem = document.getElementById('addElem');
+
+var liCount = document.getElementsByTagName('li');
```

Comment:
`liCount` to by był jakbyś zrobił `document.getElementsByTagName('li').length` 

swoją drogą `innerHTML` jest spoko do dodawania tekstu np samego `item + itemCount` ale tworzenie za jego pomocą `li`, `div`, większych struktur itp jest słabe i podatne na błędy. Spróbujesz użyć tutaj `document.createElement` ?